lua/ttt2/extensions/table.lua

```diff
@@ -0,0 +1,59 @@
+---
+-- table exentsions
+-- @author saibotk
+
+AddCSLuaFile()
+
+---
+-- Get the value from a table with a path that is given as a table of indexes.
+-- The method will try to traverse the dataTable and return the value or
+-- it will return nil, if a key (even a subpath) is not defined.
+--
+-- @param dataTable table the table to traverse and search the path in.
+-- @param path table the table with keys that will be used to traverse the tree (in order).
+-- @return any the value at the given path or nil if it does not exist
+function table.GetWithPath(dataTable, path)
+	assert(path, "table.GetWithPath(..) missing path parameter.")
+
+	-- Convert single key to table
+	if not istable(path) then
+		path = { path }
+	end
+
+	local currentDataTable = dataTable
```

Comment:
You are setting this the reference of another table, modifying this means modifying the other table to, so why not directly using the datatable var and why returning anything? I suggest returning a serial list if possible to improve e.g. https://github.com/TTT-2/TTT2/pull/467/files#diff-9c5d40549b43b8791131264c3caf1f89R99

Reply:
I used the local var, because it is overwritten with references to the subtables of the dataTable in the for loop below, and i did not want to magically overwrite the dataTable parameter, which seemed rather confusing